Adds new UMIPs for Optimistic Oracle, Perpetual, and EMP Upgrade (#187)
UMAprotocol/UMIPs: UMIPs/umip-53.md

| Authors    | John Shutt (john@umaproject.org) |
| Status     | Draft                                                                                                                                    |
| Created    | February 16, 2021                                                                                                                           |

## Summary (2-5 sentences)

This UMIP registers the `Perpetual` template with the DVM. This financial contract template enables the creation of non-expiring priceless synthetic tokens, through the use of the optimistic oracle and a funding rate mechanism.

# Motivation & Rationale

The Perpetual contract template will allow for the creation of non-expiring synthetic assets, which are useful for a wide variety of use cases. A simple example would be a synthetic asset that tracks the price of a stock index without an expiration date. This can be easier for token holders, token sponsors, and liquidity providers to manage, as it does not require them to roll over to a new token on some expiration date.

# Technical Specification

> 

Reviewer: Just copying deployments content from the other new financial contract UMIPs.

```suggestion
To accomplish this upgrade, a new financial contract template must be deployed.

After deployment, this new `PerpetualCreator` contract should be approved as a ContractCreator in the Registry. 

Note: the voters only vote for the approval -- the deployment has no effect on the protocol until added as a ContractCreator.
```

- string syntheticSymbol
- FixedPoint.Unsigned collateralRequirement
- FixedPoint.Unsigned disputeBondPercentage
- FixedPoint.Unsigned sponsorDisputeRewardPercentage
- FixedPoint.Unsigned disputerDisputeRewardPercentage
- FixedPoint.Unsigned minSponsorTokens
- FixedPoint.Unsigned tokenScaling
- uint256 withdrawalLiveness
- uint256 liquidationLiveness

Deploying a perpetual synthetic token also requires deploying a `ConfigStore` with the following parameters.

- uint256 timelockLiveness (Liveness period (in seconds) for an update to currentConfig to become official.)
- FixedPoint.Unsigned rewardRatePerSecond (Reward rate paid to successful proposers. Percentage of 1 E.g., .1 is 10%.)
- FixedPoint.Unsigned proposerBondPercentage (Bond % (of given contract's PfC) that must be staked by proposers. Percentage of 1, e.g. 0.0005 is 0.05%.)
- FixedPoint.Signed maxFundingRate (Maximum funding rate % per second that can be proposed.)
- FixedPoint.Signed minFundingRate (Minimum funding rate % per second that can be proposed.)
- uint256 proposalTimePastLimit (Funding rate proposal timestamp cannot be more than this amount of seconds in the past from the latest update time.)

# Implementation

> Please see this [directory](https://github.com/UMAprotocol/protocol/tree/master/core/contracts/financial-templates/implementation).

Reviewer: ```suggestion
Please see this [directory](https://github.com/UMAprotocol/protocol/tree/master/packages/core/contracts/financial-templates/perpetual-multiparty).
```

> The directory contains both the [implementation](https://github.com/UMAprotocol/protocol/blob/master/core/contracts/financial-templates/perpetual-multi-party/implementation/Perpetual.sol) of the `Perpetual` template and the [deployer contract](https://github.com/UMAprotocol/protocol/blob/master/core/contracts/financial-templates/perpetual-multi-party/implementation/PerpetualCreator.sol) that will be registered with the DVM to allow users to deploy their own `Perpetual` contract.

Reviewer: ```suggestion
The directory contains both the [implementation](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/perpetual-multiparty/Perpetual.sol) of the `Perpetual` template and the [deployer contract](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol) that will be registered with the DVM to allow users to deploy their own `Perpetual` contract.
```
